### PR TITLE
add new PASS filtering rules for kubelet access to fix kubelet access

### PR DESCRIPTION
**Pull request by kartoch**
Proposal for ovh/terraform-ovh-publiccloud-k8s#65 :
- not tested
- private cloud configuration has not been changed.

### CODE REVIEW

**Comment by kartoch** [comment]
Second item is related to #68 and is tested.

**Comment by kartoch** [comment]
Second and third items are related to #68 and are tested.

File: examples/public-cluster-cl-prebuilt/main.tf [hunk 1]
```diff
@@ -37,6 +37,26 @@ resource "openstack_networking_secgroup_rule_v2" "in_traffic_ssh_master" {
...
+  port_range_min    = 10250
+  port_range_max    = 10250
+  security_group_id = "${module.k8s_secgroups.master_group_id}"
+}
+
+resource "openstack_networking_secgroup_rule_v2" "k8s-kubelet-workers" {
+  direction         = "ingress"
+  ethertype         = "IPv4"
+  protocol          = "tcp"
+  remote_ip_prefix = "${data.http.myip.body}/32"
```
Comment (yanndegat) [inline examples/public-cluster-cl-prebuilt/main.tf]: why is communication with the CLI required on port 10250?

File: examples/public-cluster-cl/main.tf [hunk 2]
```diff
@@ -37,6 +37,26 @@ resource "openstack_networking_secgroup_rule_v2" "in_traffic_ssh_master" {
...
+  port_range_min    = 10250
+  port_range_max    = 10250
+  security_group_id = "${module.k8s_secgroups.master_group_id}"
+}
+
+resource "openstack_networking_secgroup_rule_v2" "k8s-kubelet-workers" {
+  direction         = "ingress"
+  ethertype         = "IPv4"
+  protocol          = "tcp"
+  remote_ip_prefix = "${data.http.myip.body}/32"
```
Comment (yanndegat) [inline examples/public-cluster-cl/main.tf]: same here

File: examples/public-cluster-cl/main.tf [hunk 3]
```diff
@@ -37,6 +37,26 @@ resource "openstack_networking_secgroup_rule_v2" "in_traffic_ssh_master" {
   security_group_id = "${module.k8s_secgroups.master_group_id}"
 }
 
+resource "openstack_networking_secgroup_rule_v2" "k8s-kubelet-masters" {
+  direction         = "ingress"
+  ethertype         = "IPv4"
+  protocol          = "tcp"
+  remote_ip_prefix = "${data.http.myip.body}/32"
```
Comment (yanndegat) [inline examples/public-cluster-cl/main.tf]: same here

File: examples/public-cluster-with-workers-cl/main.tf [hunk 4]
```diff
@@ -46,6 +46,27 @@ resource "openstack_networking_secgroup_rule_v2" "k8s-api" {
...
+  port_range_min    = 10250
+  port_range_max    = 10250
+  security_group_id = "${module.k8s_secgroups.master_group_id}"
+}
+
+resource "openstack_networking_secgroup_rule_v2" "k8s-kubelet-workers" {
+  direction         = "ingress"
+  ethertype         = "IPv4"
+  protocol          = "tcp"
+  remote_ip_prefix = "${data.http.myip.body}/32"
```
Comment (yanndegat) [inline examples/public-cluster-with-workers-cl/main.tf]: same here

**Comment by yanndegat** [comment]
i don't understand why comm with the CLI has to be enabled on port 10250/10255 ?
i thought only 6443 should be opened.

**Comment by kartoch** [comment]
For a strange reason, during tests logs were not available if these rules were not present. I've just remove the rules from horizon and it is working now (?!). So only the rules between masters and workers/masters are needed it seems.

**Comment by yanndegat** [comment]
super!

thanks a lot

can you squash the commits before merge?

**Comment by kartoch** [comment]
Done, closing #65 and #68

**Comment by yanndegat** [comment]
thanks again

**Pull request merged**